### PR TITLE
Refactor/legacy client

### PR DESCRIPTION
**Pull request by brunoMarchiEncora**
# Description

Removed some code that was not being used, renamed base packages and refactored code to leverage more spring boot and spring data functionalities.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

No new testes were needed

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

This PR requires #65 to be merged first

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[API](https://nr-forest-client-api-66-backend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-forest-client-api/actions/workflows/merge-main.yml)

### CODE REVIEW

File: src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package ca.bc.gov.api.oracle.legacy.dto;
+
+/**
+ * @author Maria Martinez, Government of BC
+ * @version 1.0.0
+ * @created 2022-11-03
+ */
+
+public class ClientPublicFilterObjectVO {
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java:9]: This class can be changed to record

Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java:9]: Change the name to end with **Dto**, to keep the naming convention

File: src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@RequiredArgsConstructor
+public class ClientPublicViewDto {
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java:12]: This class can be changed to record

File: src/main/java/ca/bc/gov/api/oracle/legacy/entity/ClientPublicViewEntity.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.util.Objects;
+import org.springframework.stereotype.Component;
+
+@Entity
+@Table(name = "V_CLIENT_PUBLIC", schema = OraclePersistenceConfiguration.ORACLE_ATTRIBUTE_SCHEMA)
+@Component
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/entity/ClientPublicViewEntity.java:13]: Remove component from the entity

Comment (paulushcgcj) [reply src/main/java/ca/bc/gov/api/oracle/legacy/entity/ClientPublicViewEntity.java:13]: Add lombok annotations

File: src/main/java/ca/bc/gov/api/oracle/legacy/util/ClientMapper.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package ca.bc.gov.api.oracle.legacy.util;
+
+import ca.bc.gov.api.oracle.legacy.dto.ClientPublicViewDto;
+import ca.bc.gov.api.oracle.legacy.entity.ClientPublicViewEntity;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ClientMapper {
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/util/ClientMapper.java:9]: Maybe we can plan on including a mapper dependency such as orika to deal with Dto <-> Entity conversions

**Review by paulushcgcj — CHANGES_REQUESTED** [review]

File: src/main/java/ca/bc/gov/api/oracle/legacy/configuration/OraclePersistenceConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    return DataSourceBuilder.create().build();
+  }
+
+  @Bean(name = "oracleEntityManager")
+  public LocalContainerEntityManagerFactoryBean oracleEntityManager(
+      final EntityManagerFactoryBuilder builder,
+      @Qualifier("oracleDataSource") final DataSource dataSource) {
+    return builder.dataSource(dataSource)
+        .packages("ca.bc.gov.api.m.oracle")
+        .persistenceUnit("oracle")
```
[the remark this answers is not in the document]
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/oracle/legacy/configuration/OraclePersistenceConfiguration.java:36]: Class will be deleted, this configuration can be done through application.yml

File: src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java (hunk 1 above)
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java:9]: Class is not being used so it's deleted.

Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicFilterObjectVO.java:9]: Class deleted

File: src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package ca.bc.gov.api.oracle.legacy.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
```
Comment (paulushcgcj) [inline src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java:8]: You can remove lombok annotations

**Review by paulushcgcj — CHANGES_REQUESTED** [review]

File: src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java (hunk 6 above)
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/oracle/legacy/dto/ClientPublicViewDto.java]: Removed lombok annotations but left @Data, should it be removed as well?

File: src/main/java/ca/bc/gov/api/oracle/legacy/util/ClientMapper.java (hunk 4 above)
Comment (brunoMarchiEncora) [reply src/main/java/ca/bc/gov/api/oracle/legacy/util/ClientMapper.java:9]: Will be done in future PR, when moving to webflux

File: pom.xml [hunk 7]
```diff
@@ -4,99 +4,412 @@
...
 		<dependency>
 			<groupId>com.h2database</groupId>
 			<artifactId>h2</artifactId>
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
+			<groupId>com.querydsl</groupId>
```
[the remark this answers is not in the document]
Comment (brunoMarchiEncora) [reply pom.xml:71]: No, removed unused dependency

**Review by paulushcgcj — APPROVED** [review]
Just rebase and go

**Comment by MCatherine1994** [comment]
Error got from OpenShift deployment:
image

**Comment by paulushcgcj** [comment]
> Error got from OpenShift deployment: image

We are fixing it right now

**Comment by MCatherine1994** [comment]
> > Error got from OpenShift deployment: image
> 
> We are fixing it right now

Thanks! Looks good now! The new api path also looks good to me!

File: src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  public ResponseEntity<List<ClientPublicViewDto>> findAllNonIndividuals(
+      @RequestParam(defaultValue = "1") int currentPage,
+      @RequestParam(defaultValue = "10") int itemsPerPage,
+      @RequestParam(defaultValue = "clientName") String sortedColumnName) {
+
+    return ResponseEntity.ok(
+        clientService.findAllNonIndividualClients(currentPage, itemsPerPage, sortedColumnName));
+  }
+
+  @GetMapping(value = "/search")
```
Comment (MCatherine1994) [inline src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java:40]: maybe we should call it "/findByNames"? I feel call it "/search" is too general, I feel search means could search based on anything

**Review by mamartinezmejia — APPROVED** [review]

File: src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java (hunk 8 above)
Comment (mamartinezmejia) [reply src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java:40]: Yes, I agree with Catherine!

Comment (paulushcgcj) [reply src/main/java/ca/bc/gov/api/oracle/legacy/controller/ClientController.java:40]: We will change the name of the endpoint in the next iteration

**Review by MCatherine1994 — APPROVED** [review]

**Pull request merged**